src/control/server/interceptors.go

```diff
@@ -116,7 +121,25 @@ var selfServerComponent = func() *build.VersionedComponent {
 	return self
 }()
 
-func checkVersion(ctx context.Context, self *build.VersionedComponent, req interface{}) error {
+func compVersionFromHeaders(ctx context.Context) (*build.VersionedComponent, error) {
+	md, hasMD := metadata.FromIncomingContext(ctx)
+	if !hasMD {
+		return nil, errNoReqMetadata
+	}
+	compName, hasName := md[proto.DaosComponentHeader]
+	if !hasName {
+		return nil, errNoReqMetadata
+	}
+	comp := build.Component(compName[0])
+	compVersion, hasVersion := md[proto.DaosVersionHeader]
+	if !hasVersion {
+		return nil, errNoReqMetadata
```

Comment:
is there value in wrapping these generic errors to tell which part of the meta data is missing? might be helpful in diagnosing problems